python/tk_desktop/ui/desktop_window.py

```diff
@@ -17,7 +17,7 @@ def setupUi(self, DesktopWindow):
         icon.addPixmap(QtGui.QPixmap(":/tk-desktop/default_systray_icon.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
         DesktopWindow.setWindowIcon(icon)
         DesktopWindow.setDockNestingEnabled(False)
-        DesktopWindow.setUnifiedTitleAndToolBarOnMac(True)
+        DesktopWindow.setUnifiedTitleAndToolBarOnMac(False)
```

Comment:
This is what was causing the transparency issues on the docking on Mac.
Setting it to false doesn't appear to make an difference when running in PySide, but it does fix the PySide2.
Qt dock advise against using this in Qt5.
QT 5: https://doc.qt.io/qt-5/qmainwindow.html#unifiedTitleAndToolBarOnMac-prop
QT 4: https://doc.qt.io/archives/qt-4.8/qmainwindow.html#unifiedTitleAndToolBarOnMac-prop